File: leetcode/260.只出现一次的数字 III.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        xor = 0
        for num in nums:
            xor ^= num
        lb = xor & -xor
        a, b = 0, 0
        for num in nums:
            if num & lb:
                a ^= num
            else:
                b ^= num
        return [a, b]

File: leetcode/260.只出现一次的数字 III.go
func singleNumber(nums []int) []int {
    xor := 0
    for _, num := range nums {
        xor ^= num
    }
    lb := xor & -xor
    a, b := 0, 0
    for _, num := range nums {
        if num & lb > 0 {
            a ^= num
        } else {
            b ^= num
        }
    }
    return []int{a, b}
}